library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -313,9 +321,111 @@ PlatformBridgeFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
   if (status == Http::FilterTrailersStatus::StopIteration) {
     pending_response_trailers_ = &trailers;
   }
+  response_complete_ = true;
   return status;
 }
 
+void PlatformBridgeFilter::onResumeDecoding() {
+  Buffer::Instance* internal_buffer = nullptr;
+  if (decoder_callbacks_->decodingBuffer()) {
+    decoder_callbacks_->modifyDecodingBuffer([&internal_buffer](Buffer::Instance& mutable_buffer) {
+      internal_buffer = &mutable_buffer;
+    });
+  }
+
+  envoy_headers* pending_headers = nullptr;
+  envoy_data* pending_data = nullptr;
+  envoy_headers* pending_trailers = nullptr;
```

Comment:
nits on naming:
- `pending_*` here might be easily misread between `pending_headers`/`pending_request_header`. Might be worth changing to `headers_to_send` or something like that
- why `internal_buffer` instead of `pending_request_data` to align with the others?

Reply:
I now see below that `internal_buffer` is reused between request/response, which I assume is the reason for that name. Are there any concerns around that state not being properly cleared before starting to receive response data?

Reply:
`internal_buffer` is a temporary pointer to the buffer needed for that invocation. The naming here is consistent with the other invocations, which do share more code. (The actual underlying memory is never shared between the request and response path.) I elected not to share code in this case as I felt it introduced more complexity rather than simplified things, due to the number of pending references in this singular case.

But honestly no strong feelings. Since the code isn't shared, we could be more particular about the naming.